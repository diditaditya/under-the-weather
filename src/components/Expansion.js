import React from 'react';
import { View, Text, Animated, StyleSheet } from 'react-native';

class Expansion from React.Component {

  render() {
    return (
      <View>
        <Text>For Future Expansion</Text>
      </View>
    );
  }


}

export default Expansion;

//google map key AIzaSyBCnrRCfgm831tHFx-onfyoxkPO8tmUD3A
