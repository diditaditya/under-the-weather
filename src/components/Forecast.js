import React from 'react';
import { Image } from 'react-native';
import { connect } from 'react-redux';

import NavBar from './NavBar';
import Header from './Header';
import Body from './Body';

const wallpaper = require('../images/beautiful_light_blue_sky-wallpaper-480x800.jpg');

const styles = {
  container: {
    flex: 1,
    width: null,
    height: null,
  },
};

export class Forecast extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      wallpaper,
    };
  }

  componentWillMount() {
    if (this.props.wallpaper) {
      const image = require(this.props.wallpaper);
      this.setState({
        wallpaper: image,
      });
    }
  }

  render() {
    // console.log(this.props);
    return (
      <Image source={this.state.wallpaper} style={styles.container}>
        <NavBar navigation={this.props.navigation} />
        <Header
          locationData={this.props.locationData}
          OWCurrentWeather={this.props.OWCurrentWeather}
        />
        <Body />
      </Image>
    );
  }

}

function mapStateToProps(state) {
  return ({
    wallpaper: state.Geolocation.selectedWallpaper,
    locationData: state.Geolocation.locationData,
    OWCurrentWeather: state.OpenWeather.currentWeather,
  });
}

export default connect(mapStateToProps, null)(Forecast);
