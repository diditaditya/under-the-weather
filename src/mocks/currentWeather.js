export const OWCurrentWeather = {
  "coord": {
    "lon": 106.87,
    "lat": -6.18
  },
  "weather": [
    {
      "id": 802,
      "main": "Clouds",
      "description": "scattered clouds",
      "icon": "03d"
    }
  ],
  "base": "stations",
  "main": {
    "temp": 30,
    "pressure": 1010,
    "humidity": 70,
    "temp_min": 30,
    "temp_max": 30
  },
  "visibility": 8000,
  "wind": {
    "speed": 2.6
  },
  "clouds": {
    "all": 40
  },
  "dt": 1496379600,
  "sys": {
    "type": 1,
    "id": 8043,
    "message": 0.0132,
    "country": "ID",
    "sunrise": 1496357821,
    "sunset": 1496400242
  },
  "id": 1642907,
  "name": "Daerah Khusus Ibukota Jakarta",
  "cod": 200
}
