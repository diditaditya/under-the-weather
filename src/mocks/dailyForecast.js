export const dailyForecast = {
  "city": {
    "id": 1642907,
    "name": "Daerah Khusus Ibukota Jakarta",
    "coord": {
      "lon": 106.8584,
      "lat": -6.2183
    },
    "country": "ID",
    "population": 0
  },
  "cod": "200",
  "message": 0.0559456,
  "cnt": 10,
  "list": [
    {
      "dt": 1496376000,
      "temp": {
        "day": 30,
        "min": 28.79,
        "max": 31.47,
        "night": 28.79,
        "eve": 31.36,
        "morn": 30
      },
      "pressure": 1025.02,
      "humidity": 91,
      "weather": [
        {
          "id": 803,
          "main": "Clouds",
          "description": "broken clouds",
          "icon": "04d"
        }
      ],
      "speed": 4.56,
      "deg": 104,
      "clouds": 68
    }, {
      "dt": 1496462400,
      "temp": {
        "day": 29.88,
        "min": 27.74,
        "max": 31.02,
        "night": 28.44,
        "eve": 30.78,
        "morn": 27.74
      },
      "pressure": 1024.73,
      "humidity": 89,
      "weather": [
        {
          "id": 500,
          "main": "Rain",
          "description": "light rain",
          "icon": "10d"
        }
      ],
      "speed": 3.46,
      "deg": 91,
      "clouds": 0,
      "rain": 1.27
    }, {
      "dt": 1496548800,
      "temp": {
        "day": 28.98,
        "min": 26.93,
        "max": 30.09,
        "night": 27.53,
        "eve": 29.41,
        "morn": 26.93
      },
      "pressure": 1024.18,
      "humidity": 95,
      "weather": [
        {
          "id": 500,
          "main": "Rain",
          "description": "light rain",
          "icon": "10d"
        }
      ],
      "speed": 1.52,
      "deg": 223,
      "clouds": 0,
      "rain": 1.05
    }, {
      "dt": 1496635200,
      "temp": {
        "day": 30.04,
        "min": 26.33,
        "max": 30.04,
        "night": 26.92,
        "eve": 28.26,
        "morn": 26.33
      },
      "pressure": 1010.58,
      "humidity": 0,
      "weather": [
        {
          "id": 500,
          "main": "Rain",
          "description": "light rain",
          "icon": "10d"
        }
      ],
      "speed": 1.32,
      "deg": 9,
      "clouds": 12,
      "rain": 1.16
    }, {
      "dt": 1496721600,
      "temp": {
        "day": 30.19,
        "min": 26.82,
        "max": 30.19,
        "night": 26.82,
        "eve": 28.16,
        "morn": 26.89
      },
      "pressure": 1011.35,
      "humidity": 0,
      "weather": [
        {
          "id": 500,
          "main": "Rain",
          "description": "light rain",
          "icon": "10d"
        }
      ],
      "speed": 1.79,
      "deg": 43,
      "clouds": 29,
      "rain": 1.17
    }, {
      "dt": 1496808000,
      "temp": {
        "day": 30.14,
        "min": 26.49,
        "max": 30.14,
        "night": 26.49,
        "eve": 27.61,
        "morn": 26.89
      },
      "pressure": 1010.73,
      "humidity": 0,
      "weather": [
        {
          "id": 500,
          "main": "Rain",
          "description": "light rain",
          "icon": "10d"
        }
      ],
      "speed": 3.92,
      "deg": 82,
      "clouds": 18,
      "rain": 1.56
    }, {
      "dt": 1496894400,
      "temp": {
        "day": 29.13,
        "min": 26.25,
        "max": 29.13,
        "night": 26.25,
        "eve": 27.06,
        "morn": 26.5
      },
      "pressure": 1010.32,
      "humidity": 0,
      "weather": [
        {
          "id": 501,
          "main": "Rain",
          "description": "moderate rain",
          "icon": "10d"
        }
      ],
      "speed": 3.21,
      "deg": 82,
      "clouds": 42,
      "rain": 6.27
    }, {
      "dt": 1496980800,
      "temp": {
        "day": 29.61,
        "min": 26.28,
        "max": 29.61,
        "night": 26.43,
        "eve": 27.56,
        "morn": 26.28
      },
      "pressure": 1009.72,
      "humidity": 0,
      "weather": [
        {
          "id": 500,
          "main": "Rain",
          "description": "light rain",
          "icon": "10d"
        }
      ],
      "speed": 5.26,
      "deg": 75,
      "clouds": 27,
      "rain": 1.82
    }, {
      "dt": 1497067200,
      "temp": {
        "day": 29.25,
        "min": 26.01,
        "max": 29.25,
        "night": 26.28,
        "eve": 27.34,
        "morn": 26.01
      },
      "pressure": 1010.2,
      "humidity": 0,
      "weather": [
        {
          "id": 500,
          "main": "Rain",
          "description": "light rain",
          "icon": "10d"
        }
      ],
      "speed": 3.74,
      "deg": 76,
      "clouds": 55,
      "rain": 2.21
    }, {
      "dt": 1497153600,
      "temp": {
        "day": 28.98,
        "min": 25.55,
        "max": 28.98,
        "night": 25.55,
        "eve": 26.89,
        "morn": 26.09
      },
      "pressure": 1009.62,
      "humidity": 0,
      "weather": [
        {
          "id": 501,
          "main": "Rain",
          "description": "moderate rain",
          "icon": "10d"
        }
      ],
      "speed": 5.09,
      "deg": 57,
      "clouds": 37,
      "rain": 3.86
    }
  ]
}
