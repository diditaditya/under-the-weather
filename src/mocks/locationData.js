export const locationData = {
  "results": [
    {
      "address_components": [
        {
          "long_name": "16",
          "short_name": "16",
          "types": ["street_number"]
        }, {
          "long_name": "Jalan Cempaka Putih Tengah XV",
          "short_name": "Jl. Cempaka Putih Tengah XV",
          "types": ["route"]
        }, {
          "long_name": "",
          "short_name": "",
          "types": ["political"]
        }, {
          "long_name": "",
          "short_name": "",
          "types": ["political"]
        }, {
          "long_name": "Cempaka Putih Timur",
          "short_name": "Cemp. Putih Tim.",
          "types": ["administrative_area_level_4", "political"]
        }, {
          "long_name": "Cempaka Putih",
          "short_name": "Cemp. Putih",
          "types": ["administrative_area_level_3", "political"]
        }, {
          "long_name": "Kota Jakarta Pusat",
          "short_name": "Kota Jakarta Pusat",
          "types": ["administrative_area_level_2", "political"]
        }, {
          "long_name": "Daerah Khusus Ibukota Jakarta",
          "short_name": "Daerah Khusus Ibukota Jakarta",
          "types": ["administrative_area_level_1", "political"]
        }, {
          "long_name": "Indonesia",
          "short_name": "ID",
          "types": ["country", "political"]
        }, {
          "long_name": "10510",
          "short_name": "10510",
          "types": ["postal_code"]
        }
      ],
      "formatted_address": "Jl. Cempaka Putih Tengah XV No.16, Cemp. Putih Tim., Cemp. Putih, Kota Jakarta Pusat, Daerah Khusus Ibukota Jakarta 10510, Indonesia",
      "geometry": {
        "location": {
          "lat": -6.175078,
          "lng": 106.865087
        },
        "location_type": "ROOFTOP",
        "viewport": {
          "northeast": {
            "lat": -6.173729019708498,
            "lng": 106.8664359802915
          },
          "southwest": {
            "lat": -6.176426980291502,
            "lng": 106.8637380197085
          }
        }
      },
      "place_id": "ChIJ1dUaAAD1aS4RA4a_-bFagCE",
      "types": ["street_address"]
    }, {
      "address_components": [
        {
          "long_name": "Central Jakarta",
          "short_name": "Central Jakarta",
          "types": ["locality", "political"]
        }, {
          "long_name": "Central Jakarta City",
          "short_name": "Central Jakarta City",
          "types": ["administrative_area_level_2", "political"]
        }, {
          "long_name": "Jakarta",
          "short_name": "Jakarta",
          "types": ["administrative_area_level_1", "political"]
        }, {
          "long_name": "Indonesia",
          "short_name": "ID",
          "types": ["country", "political"]
        }
      ],
      "formatted_address": "Central Jakarta, Central Jakarta City, Jakarta, Indonesia",
      "geometry": {
        "bounds": {
          "northeast": {
            "lat": -6.135320999999999,
            "lng": 106.8810121
          },
          "southwest": {
            "lat": -6.229305,
            "lng": 106.7919172
          }
        },
        "location": {
          "lat": -6.1864864,
          "lng": 106.8340911
        },
        "location_type": "APPROXIMATE",
        "viewport": {
          "northeast": {
            "lat": -6.135320999999999,
            "lng": 106.8810121
          },
          "southwest": {
            "lat": -6.229305,
            "lng": 106.7919201
          }
        }
      },
      "place_id": "ChIJB0vJuDb0aS4R9oJ8iznVpm4",
      "types": ["locality", "political"]
    }, {
      "address_components": [
        {
          "long_name": "Jakarta",
          "short_name": "Jakarta",
          "types": ["colloquial_area", "locality", "political"]
        }, {
          "long_name": "Jakarta",
          "short_name": "Jakarta",
          "types": ["administrative_area_level_1", "political"]
        }, {
          "long_name": "Indonesia",
          "short_name": "ID",
          "types": ["country", "political"]
        }
      ],
      "formatted_address": "Jakarta, Indonesia",
      "geometry": {
        "bounds": {
          "northeast": {
            "lat": -6.0886599,
            "lng": 106.972825
          },
          "southwest": {
            "lat": -6.3708331,
            "lng": 106.686211
          }
        },
        "location": {
          "lat": -6.2087634,
          "lng": 106.845599
        },
        "location_type": "APPROXIMATE",
        "viewport": {
          "northeast": {
            "lat": -6.0886599,
            "lng": 106.972825
          },
          "southwest": {
            "lat": -6.3708331,
            "lng": 106.686211
          }
        }
      },
      "place_id": "ChIJnUvjRenzaS4RoobX2g-_cVM",
      "types": ["colloquial_area", "locality", "political"]
    }, {
      "address_components": [
        {
          "long_name": "2",
          "short_name": "2",
          "types": ["political"]
        }, {
          "long_name": "",
          "short_name": "",
          "types": ["political"]
        }, {
          "long_name": "Cempaka Putih Timur",
          "short_name": "Cemp. Putih Tim.",
          "types": ["administrative_area_level_4", "political"]
        }, {
          "long_name": "Cempaka Putih",
          "short_name": "Cemp. Putih",
          "types": ["administrative_area_level_3", "political"]
        }, {
          "long_name": "Kota Jakarta Pusat",
          "short_name": "Kota Jakarta Pusat",
          "types": ["administrative_area_level_2", "political"]
        }, {
          "long_name": "Daerah Khusus Ibukota Jakarta",
          "short_name": "Daerah Khusus Ibukota Jakarta",
          "types": ["administrative_area_level_1", "political"]
        }, {
          "long_name": "Indonesia",
          "short_name": "ID",
          "types": ["country", "political"]
        }
      ],
      "formatted_address": "RT.2, Cemp. Putih Tim., Cemp. Putih, Kota Jakarta Pusat, Daerah Khusus Ibukota Jakarta, Indonesia",
      "geometry": {
        "bounds": {
          "northeast": {
            "lat": -6.173481,
            "lng": 106.8679419
          },
          "southwest": {
            "lat": -6.176254000000001,
            "lng": 106.864878
          }
        },
        "location": {
          "lat": -6.1754171,
          "lng": 106.8654083
        },
        "location_type": "APPROXIMATE",
        "viewport": {
          "northeast": {
            "lat": -6.173481,
            "lng": 106.8679419
          },
          "southwest": {
            "lat": -6.176254000000001,
            "lng": 106.864878
          }
        }
      },
      "place_id": "ChIJ2YMNHwD1aS4R96wzCawb0yE",
      "types": ["political"]
    }, {
      "address_components": [
        {
          "long_name": "8",
          "short_name": "8",
          "types": ["political"]
        }, {
          "long_name": "Cempaka Putih Timur",
          "short_name": "Cemp. Putih Tim.",
          "types": ["administrative_area_level_4", "political"]
        }, {
          "long_name": "Cempaka Putih",
          "short_name": "Cemp. Putih",
          "types": ["administrative_area_level_3", "political"]
        }, {
          "long_name": "Kota Jakarta Pusat",
          "short_name": "Kota Jakarta Pusat",
          "types": ["administrative_area_level_2", "political"]
        }, {
          "long_name": "Daerah Khusus Ibukota Jakarta",
          "short_name": "Daerah Khusus Ibukota Jakarta",
          "types": ["administrative_area_level_1", "political"]
        }, {
          "long_name": "Indonesia",
          "short_name": "ID",
          "types": ["country", "political"]
        }
      ],
      "formatted_address": "RW.8, Cemp. Putih Tim., Cemp. Putih, Kota Jakarta Pusat, Daerah Khusus Ibukota Jakarta, Indonesia",
      "geometry": {
        "bounds": {
          "northeast": {
            "lat": -6.1729039,
            "lng": 106.8721418
          },
          "southwest": {
            "lat": -6.176314,
            "lng": 106.864153
          }
        },
        "location": {
          "lat": -6.1755806,
          "lng": 106.8702028
        },
        "location_type": "APPROXIMATE",
        "viewport": {
          "northeast": {
            "lat": -6.1729039,
            "lng": 106.8721418
          },
          "southwest": {
            "lat": -6.176314,
            "lng": 106.864153
          }
        }
      },
      "place_id": "ChIJDbEM0wH1aS4R2SnRob7aLs0",
      "types": ["political"]
    }, {
      "address_components": [
        {
          "long_name": "10510",
          "short_name": "10510",
          "types": ["postal_code"]
        }, {
          "long_name": "East Cempaka Putih",
          "short_name": "East Cempaka Putih",
          "types": ["administrative_area_level_4", "political"]
        }, {
          "long_name": "Cempaka Putih",
          "short_name": "Cempaka Putih",
          "types": ["administrative_area_level_3", "political"]
        }, {
          "long_name": "Central Jakarta City",
          "short_name": "Central Jakarta City",
          "types": ["administrative_area_level_2", "political"]
        }, {
          "long_name": "Jakarta",
          "short_name": "Jakarta",
          "types": ["administrative_area_level_1", "political"]
        }, {
          "long_name": "Indonesia",
          "short_name": "ID",
          "types": ["country", "political"]
        }
      ],
      "formatted_address": "East Cempaka Putih, Cempaka Putih, Central Jakarta City, Jakarta 10510, Indonesia",
      "geometry": {
        "bounds": {
          "northeast": {
            "lat": -6.1663198,
            "lng": 106.8790512
          },
          "southwest": {
            "lat": -6.188108400000001,
            "lng": 106.8644485
          }
        },
        "location": {
          "lat": -6.1761468,
          "lng": 106.8724156
        },
        "location_type": "APPROXIMATE",
        "viewport": {
          "northeast": {
            "lat": -6.1663198,
            "lng": 106.8790512
          },
          "southwest": {
            "lat": -6.188108400000001,
            "lng": 106.8644485
          }
        }
      },
      "place_id": "ChIJm-SGA-P0aS4RsGQDTevFABw",
      "types": ["postal_code"]
    }, {
      "address_components": [
        {
          "long_name": "East Cempaka Putih",
          "short_name": "East Cempaka Putih",
          "types": ["administrative_area_level_4", "political"]
        }, {
          "long_name": "Cempaka Putih",
          "short_name": "Cempaka Putih",
          "types": ["administrative_area_level_3", "political"]
        }, {
          "long_name": "Central Jakarta City",
          "short_name": "Central Jakarta City",
          "types": ["administrative_area_level_2", "political"]
        }, {
          "long_name": "Jakarta",
          "short_name": "Jakarta",
          "types": ["administrative_area_level_1", "political"]
        }, {
          "long_name": "Indonesia",
          "short_name": "ID",
          "types": ["country", "political"]
        }
      ],
      "formatted_address": "East Cempaka Putih, Cempaka Putih, Central Jakarta City, Jakarta, Indonesia",
      "geometry": {
        "bounds": {
          "northeast": {
            "lat": -6.1663799,
            "lng": 106.878807
          },
          "southwest": {
            "lat": -6.188168000000001,
            "lng": 106.8642099
          }
        },
        "location": {
          "lat": -6.1761468,
          "lng": 106.8724156
        },
        "location_type": "APPROXIMATE",
        "viewport": {
          "northeast": {
            "lat": -6.1663799,
            "lng": 106.878807
          },
          "southwest": {
            "lat": -6.188168000000001,
            "lng": 106.8642099
          }
        }
      },
      "place_id": "ChIJM3dU0v30aS4Rx3RI4iVSPdI",
      "types": ["administrative_area_level_4", "political"]
    }, {
      "address_components": [
        {
          "long_name": "Cempaka Putih",
          "short_name": "Cempaka Putih",
          "types": ["administrative_area_level_3", "political"]
        }, {
          "long_name": "Central Jakarta City",
          "short_name": "Central Jakarta City",
          "types": ["administrative_area_level_2", "political"]
        }, {
          "long_name": "Jakarta",
          "short_name": "Jakarta",
          "types": ["administrative_area_level_1", "political"]
        }, {
          "long_name": "Indonesia",
          "short_name": "ID",
          "types": ["country", "political"]
        }
      ],
      "formatted_address": "Cempaka Putih, Central Jakarta City, Jakarta, Indonesia",
      "geometry": {
        "bounds": {
          "northeast": {
            "lat": -6.1663799,
            "lng": 106.878807
          },
          "southwest": {
            "lat": -6.196158,
            "lng": 106.85534
          }
        },
        "location": {
          "lat": -6.182670799999999,
          "lng": 106.8679899
        },
        "location_type": "APPROXIMATE",
        "viewport": {
          "northeast": {
            "lat": -6.1663799,
            "lng": 106.878807
          },
          "southwest": {
            "lat": -6.196158,
            "lng": 106.85534
          }
        }
      },
      "place_id": "ChIJc-r_wvj0aS4Ru4tkAxqdPIs",
      "types": ["administrative_area_level_3", "political"]
    }, {
      "address_components": [
        {
          "long_name": "Central Jakarta City",
          "short_name": "Central Jakarta City",
          "types": ["administrative_area_level_2", "political"]
        }, {
          "long_name": "Jakarta",
          "short_name": "Jakarta",
          "types": ["administrative_area_level_1", "political"]
        }, {
          "long_name": "Indonesia",
          "short_name": "ID",
          "types": ["country", "political"]
        }
      ],
      "formatted_address": "Central Jakarta City, Jakarta, Indonesia",
      "geometry": {
        "bounds": {
          "northeast": {
            "lat": -6.135320999999999,
            "lng": 106.881432
          },
          "southwest": {
            "lat": -6.229305,
            "lng": 106.791799
          }
        },
        "location": {
          "lat": -6.1864674,
          "lng": 106.8296372
        },
        "location_type": "APPROXIMATE",
        "viewport": {
          "northeast": {
            "lat": -6.135320999999999,
            "lng": 106.881432
          },
          "southwest": {
            "lat": -6.229305,
            "lng": 106.791799
          }
        }
      },
      "place_id": "ChIJB0vJuDb0aS4RlHPzGrFt85A",
      "types": ["administrative_area_level_2", "political"]
    }, {
      "address_components": [
        {
          "long_name": "Jakarta",
          "short_name": "Jakarta",
          "types": ["administrative_area_level_1", "political"]
        }, {
          "long_name": "Indonesia",
          "short_name": "ID",
          "types": ["country", "political"]
        }
      ],
      "formatted_address": "Jakarta, Indonesia",
      "geometry": {
        "bounds": {
          "northeast": {
            "lat": -5.1843219,
            "lng": 106.972825
          },
          "southwest": {
            "lat": -6.3708331,
            "lng": 106.3831259
          }
        },
        "location": {
          "lat": -6.17511,
          "lng": 106.8650395
        },
        "location_type": "APPROXIMATE",
        "viewport": {
          "northeast": {
            "lat": -6.0886599,
            "lng": 106.972825
          },
          "southwest": {
            "lat": -6.370829899999999,
            "lng": 106.686211
          }
        }
      },
      "place_id": "ChIJnUvjRenzaS4RILjULejFAAE",
      "types": ["administrative_area_level_1", "political"]
    }
  ],
  "status": "OK"
}
