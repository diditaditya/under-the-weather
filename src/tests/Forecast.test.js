import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';

import store from '../store/configureStore';

import { Forecast } from '../components/Forecast';

import { locationData } from '../mocks/locationData';
import { OWCurrentWeather } from '../mocks/currentWeather';
import { dailyForecast } from '../mocks/dailyForecast';
import { weather } from '../mocks/weather';

const wallpaper = require('../../images/005e3efe1a23e55f2327eedd932c7742.jpg');

const getCurrentLocation = () => {}
const fetchOWCurrentWeather = () => {}

describe('Forecast', () => {

  it('should render header with location data and current weather', () => {
    const ForecastScreen = renderer.create(
      <Provider store={store}>
        <Forecast locationData={locationData} OWCurrentWeather={OWCurrentWeather} dailyForecast={''}/>
      </Provider>
    );
      expect(ForecastScreen).toMatchSnapshot();
  });

  it('should render body but loading data because of empty forecast data', () => {
    const ForecastScreen = renderer.create(
      <Provider store={store}>
        <Forecast locationData={locationData} OWCurrentWeather={OWCurrentWeather} dailyForecast={''}/>
      </Provider>
    );
      expect(ForecastScreen).toMatchSnapshot();
  });

  it('should render header and body correctly', () => {
    const ForecastScreen = renderer.create(
      <Provider store={store}>
        <Forecast locationData={locationData} OWCurrentWeather={OWCurrentWeather} dailyForecast={weather}/>
      </Provider>
    );
    expect(ForecastScreen).toMatchSnapshot();
  });

});
