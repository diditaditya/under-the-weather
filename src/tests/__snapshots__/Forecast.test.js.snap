// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forecast should render body but loading data because of empty forecast data 1`] = `
<Image
  source={1}
  style={
    Object {
      "flex": 1,
      "height": null,
      "width": null,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-around",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Home
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Daily
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Hourly
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "padding": 5,
        "width": "100%",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 25,
        }
      }
    >
      Jakarta, Indonesia
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 18,
        }
      }
    >
      30
       C
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="#999999"
      hidesWhenStopped={true}
      size="large"
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Loading Weather Forecast ..
    </Text>
  </View>
</Image>
`;

exports[`Forecast should render header and body correctly 1`] = `
<Image
  source={1}
  style={
    Object {
      "flex": 1,
      "height": null,
      "width": null,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-around",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Home
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Daily
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Hourly
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "padding": 5,
        "width": "100%",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 25,
        }
      }
    >
      Jakarta, Indonesia
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 18,
        }
      }
    >
      30
       C
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="#999999"
      hidesWhenStopped={true}
      size="large"
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Loading Weather Forecast ..
    </Text>
  </View>
</Image>
`;

exports[`Forecast should render header with location data and current weather 1`] = `
<Image
  source={1}
  style={
    Object {
      "flex": 1,
      "height": null,
      "width": null,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-around",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Home
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Daily
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Hourly
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "padding": 5,
        "width": "100%",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 25,
        }
      }
    >
      Jakarta, Indonesia
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 18,
        }
      }
    >
      30
       C
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="#999999"
      hidesWhenStopped={true}
      size="large"
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Loading Weather Forecast ..
    </Text>
  </View>
</Image>
`;
